src/qt/openuridialog.cpp

```diff
@@ -9,12 +9,15 @@
 #include <qt/sendcoinsrecipient.h>
 
 #include <QUrl>
+#include <QClipboard>
 
-OpenURIDialog::OpenURIDialog(QWidget *parent) :
+OpenURIDialog::OpenURIDialog(const PlatformStyle *platformStyle, QWidget *parent) :
     QDialog(parent),
     ui(new Ui::OpenURIDialog)
 {
     ui->setupUi(this);
+    ui->pasteButton->setIcon(platformStyle->SingleColorIcon(":/icons/editpaste"));
+    QObject::connect(ui->pasteButton, &QPushButton::clicked, [=] { ui->uriEdit->setText(QApplication::clipboard()->text()); });
```

Comment:
The `on_pasteButton_clicked()` slot with automatic connection seems simpler and more explicit.

Reply:
What's wrong with this? The lambda functions brings it on the point

Reply:
> What's wrong with this?

I did not say "wrong" ;)
Yes, it works.

> The lambda functions brings it on the point

Capturing `[=]` seems an unnecessary overhead. Traditional `on_pasteButton_clicked()` slot seems more expected for a such simple functionality.

Reply:
I kind of dislike automatic connections since it's easy to break them - for instance renaming the item.

Reply:
> I kind of dislike automatic connections since it's easy to break them - for instance renaming the item.

Make sense. Maybe just a standard connection to a named slot? (01602e453a7a731fa4a6f409cae2ffa2e31709c6 was good except old syntax).

Reply:
Agree with @hebasto
```cpp
connect(ui->pasteButton, &QPushButton::clicked, ui->uriEdit, &QLineEdit::paste);
```

Reply:
@emilengler I guess @promag's approach seems simpler... interested to implement it that way?